tools/report_build.py

```diff
@@ -1,25 +1,50 @@
 #!/usr/bin/env python3
 
+import argparse
+import copy
 import os
 import pprint
-import sys
 
 import requests
 
+from github import Github
+
+
 GITHUB_ENDPOINT = "https://api.github.com/graphql"
+GITHUB_TOKEN = os.environ["GITHUB_TOKEN"]
 GITHUB_HEADERS = {
-    "Authorization": "Bearer {}".format(os.environ["GITHUB_TOKEN"]),
+    "Authorization": "Bearer {}".format(GITHUB_TOKEN),
     "Accept": "application/vnd.github.queen-beryl-preview+json",
 }
-COMMENT_TYPES = ("IssueComment",)
+COMMENT_TYPES = ("IssueComment", "CommitComment")
 EVENT_COUNT = 50
 
 
+def dict_merge(a, b):
+    """recursively merges dict's. not just simple a['key'] = b['key'], if
+    both a and have a key whose value is a dict then dict_merge is called
+    on both values and the result stored in the returned dictionary."""
+    if not isinstance(b, dict):
+        return b
+    result = copy.deepcopy(a)
+    for k, v in b.items():
+        if k in result and isinstance(result[k], dict):
+            result[k] = dict_merge(result[k], v)
+        else:
+            result[k] = copy.deepcopy(v)
```

Comment:
Maybe unnecessary in our use case, but since this is meant to return a completely new object, it would be surprising if the values could get modified outside of this copy.

Say, we may get lists in our dict merges in the future, I'd rather not chase that bug when this happens :)

Reply:
OK, I still don't see it, but that's fine, there no real impact.